### PR TITLE
Added variable names in Models

### LINKED ISSUE #15056: Custom variable names in models
### What is the problem this feature will solve?

This would be useful for large composite models. (For example with many repeated models, where the Parameters are automatically numbered) I am not sure whether this feature already exists.

### Describe the desired outcome

something like:

from astropy.modeling.models import Gaussian1D

g = Gaussian1D(amplitude=10.2, mean=2.3, stddev=1.2, name='first')

and print(g) gives:

Model: Gaussian1D
Name: first
Inputs: ('x',)
Outputs: ('y',)
Model set size: 1
Parameters:
    first_amplitude first_mean first_stddev
    --------- ---- ------
         10.2  2.3    1.2

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by alexandrecesarba**
### Description

This PR adds customized names to existing templates. This new feature is important because it will allow users to assign more meaningful and understandable names to variables, improving the readability and comprehension of the model.

Fixes #15056

### CODE REVIEW

**Comment by pllim** [comment]
Thanks, but looks incomplete. Needs:

* Change log
* Green CI status
* User doc
* Test(s)

**Comment by alexandrecesarba** [comment]
pre-commit.ci autofix

File: astropy/modeling/core.py [hunk 1]
```diff
@@ -2877,9 +2877,13 @@ def _format_str(self, keywords=[], defaults={}):
 
         if columns:
             param_table = Table(columns, names=self.param_names)
-            # Set units on the columns
+
```
Comment (neutrinoceros) [inline astropy/modeling/core.py:2880]: I guess this comment was removed by mistake ? Can you add it back ?

Comment (alexandrecesarba) [reply astropy/modeling/core.py:2880]: Sorry! I will add it back. Thanks for noticing.

**Comment by alexandrecesarba** [comment]
pre-commit.ci autofix

**Comment by alexandrecesarba** [comment]
pre-commit.ci autofix

File: astropy/modeling/core.py [hunk 2]
```diff
@@ -2880,6 +2880,10 @@ def _format_str(self, keywords=[], defaults={}):
             # Set units on the columns
             for name in self.param_names:
                 param_table[name].unit = getattr(self, name).unit
+                if self.name is not None:
+                    param_table[name].info.name = str(self.name + "_" + name)
+                else:
+                    param_table[name].info.name = name
```
Comment (WilliamJamieson) [inline astropy/modeling/core.py:2886]:
```suggestion
                param_table[name].info.name = name if self.name is None else f"{self.name}_{name}"
```

Coverage indicates these lines are not covered by tests, pleas check that your test actually covers them.

**Review by WilliamJamieson — CHANGES_REQUESTED** [review]
This seems to have broken the doc tests. Please fix those too.

**Comment by alexandrecesarba** [comment]
> This seems to have broken the doc tests. Please fix those too.

Sorry if I misunderstood, but is there anything other than unit tests that I should do to ensure that Codecov will cover it?

**Comment by alexandrecesarba** [comment]
pre-commit.ci autofix

**Pull request closed without merging**